src/core/jbig2.js

```diff
@@ -1293,8 +1394,718 @@ var Jbig2Image = (function Jbig2ImageClosure() {
+      throw new Jbig2Error('refinement with Huffman is not supported');
+    }
+
+    return {
+      symbolIDTable,
+      tableFirstS,
+      tableDeltaS,
+      tableDeltaT,
+    };
+  }
+
+  function getSymbolDictionaryHuffmanTables(dictionary, referredTo,
+                                            customTables) {
+    // 7.4.2.1.6 Symbol dictionary segment Huffman table selection
+
+    let customIndex = 0, tableDeltaHeight, tableDeltaWidth;
+    switch (dictionary.huffmanDHSelector) {
+      case 0:
+      case 1:
+        tableDeltaHeight = getStandardTable(dictionary.huffmanDHSelector + 4);
+        break;
+      case 3:
+        tableDeltaHeight = getCustomHuffmanTable(customIndex, referredTo,
+          customTables);
+        customIndex++;
+        break;
+      default:
+        throw new Jbig2Error('invalid Huffman DH selector');
+    }
+
+    switch (dictionary.huffmanDWSelector) {
+      case 0:
+      case 1:
+        tableDeltaWidth = getStandardTable(dictionary.huffmanDWSelector + 2);
+        break;
+      case 3:
+        tableDeltaWidth = getCustomHuffmanTable(customIndex, referredTo,
+          customTables);
+        customIndex++;
+        break;
+      default:
+        throw new Jbig2Error('invalid Huffman DW selector');
+    }
+
+    let tableBitmapSize, tableAggregateInstances;
+    if (dictionary.bitmapSizeSelector) {
+      tableBitmapSize = getCustomHuffmanTable(customIndex, referredTo,
+        customTables);
+      customIndex++;
+    } else {
+      tableBitmapSize = getStandardTable(1);
+    }
+
+    if (dictionary.aggregationInstancesSelector) {
+      tableAggregateInstances = getCustomHuffmanTable(customIndex, referredTo,
+        customTables);
+    } else {
+      tableAggregateInstances = getStandardTable(1);
+    }
+
+    return {
+      tableDeltaHeight,
+      tableDeltaWidth,
+      tableBitmapSize,
+      tableAggregateInstances,
+    };
+  }
+
+  function readUncompressedBitmap(reader, width, height) {
+    let bitmap = [], x, y, row;
+    for (y = 0; y < height; y++) {
+      row = new Uint8Array(width);
+      bitmap.push(row);
+      for (x = 0; x < width; x++) {
+        row[x] = reader.readBit();
+      }
+      reader.byteAlign();
+    }
+    return bitmap;
+  }
+
+  function decodeMMRBitmap(input, width, height, endOfBlock) {
+    // MMR is the same compression algorithm as the PDF filter
+    // CCITTFaxDecode with /K -1.
+    let params = {
+      K: -1,
+      Columns: width,
+      Rows: height,
+      BlackIs1: true,
+      EndOfBlock: endOfBlock,
+    };
+    let decoder = new CCITTFaxDecoder(input, params);
+    let bitmap = [], x, y, row, currentByte, shift, eof = false;
+
+    for (y = 0; y < height; y++) {
+      row = new Uint8Array(width);
+      bitmap.push(row);
+      shift = -1;
+      for (x = 0; x < width; x++) {
+        if (shift < 0) {
+          currentByte = decoder.readNextChar();
+          if (currentByte === -1) {
+            eof = true;
```

Comment:
This is an unexpected error case and I think it's best to fill the image with zeros, which is white in JBIG2. We must continue decoding so that the number of rows in the image will be equal to `height`.